SCF-Session-4/fixXY/ArrayFixXY.java

```diff
@@ -0,0 +1,80 @@
+package fixXY;
+
+/**
+ * Contains methods to solve fixXY problem.
+ * @author Ishita Mathur
+ *
+ */
+public class ArrayFixXY 
+{
+	/**
+	 * Method returns an array that contains exactly the same numbers as the input array,
+	 * but rearranged so that every X is immediately followed by a Y.
+	 * It does not move X within array, but every other number may move.
+	 * 
+	 * @param inputArray, should contain equal number of X and Y 
+	 * & two adjacent X values should not be there & X should not be at the last position. 
+	 * @param arrayLength > 0
+	 * @param x, this is the value that should not move.
+	 * @param y, this is the value that should come after x.
+	 * @return
+	 */
+	public int[] fixXY(int[] inputArray, int arrayLength, int x, int y)
+	{
+		if (arrayLength == 0)
+			throw new AssertionError("Array is empty");
+		for(int j = 0; j < arrayLength - 1; j++)
+		{
+			if (inputArray[j] == inputArray[j + 1] && inputArray[j] == x)
+				throw new AssertionError("Two adjacent X values");
+		}
+		
+		for(int k = 0; k <= arrayLength - 1; k++)
+		{
+			if (k == arrayLength - 1 && inputArray[k] == x)
+				throw new AssertionError("X is at last position");
+		}
+
```

Comment:
Above two `for loops` are not needed to resolve this. Use the single `for loop` to resolve.

Reply:
done